WalletWasabi/Wallets/WalletManager.cs

```diff
@@ -132,9 +135,14 @@ public async Task<Wallet> StartWalletAsync(Wallet wallet)
 
 			lock (Lock)
 			{
-				if (CancelAllInitialization.IsCancellationRequested || CancelAllInitialization.Token.IsCancellationRequested)
+				if (_disposedValue)
+				{
+					throw new OperationCanceledException($"Object was already disposed.");
```

Comment:
I think that when we hit this path, it's actually a bug as it is not really acceptable to operate with a disposed object.

So maybe we can add a log line saying that we have a bug in code if we hit this path. It should be fixed in a follow up PR more robustly, I guess.